### PR TITLE
Minor fix to SPARK-6958: Improve Python docstring for DataFrame.sort.

### PR DESCRIPTION
**Pull request by rxin**
As a follow up PR to #5544.

cc @davies

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -519,7 +520,7 @@ def sort(self, *cols, **kwargs):
             jcols = [jc if asc else jc.desc()
                      for asc, jc in zip(ascending, jcols)]
         else:
-            raise TypeError("ascending can only be bool or list, but got %s" % type(ascending))
+            raise TypeError("ascending can only be boolean or list, but got %s" % type(ascending))
```
Comment (davies) [inline python/pyspark/sql/dataframe.py]: In Python, it's called `bool`

Comment (rxin) [inline python/pyspark/sql/dataframe.py]: I think bool is the name of the function, but the data type is still boolean?

see

http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.sort.html#pandas.DataFrame.sort

https://docs.python.org/2/library/stdtypes.html#boolean-values

https://docs.python.org/2/c-api/bool.html

Comment (davies) [inline python/pyspark/sql/dataframe.py]: The type for boolean is `bool`:
```
>>> type(True)
<type 'bool'>
>>> type(1)
<type 'int'>
```
Similar to `int` for integer.

Comment (davies) [inline python/pyspark/sql/dataframe.py]: It's weird to see we put `boolean` and `int` together, it should be either `boolean/integer` or `bool/int`.

Comment (rxin) [inline python/pyspark/sql/dataframe.py]: Alright I removed int, since 1, 0 has the same value as True, False anyway.

Comment (davies) [inline python/pyspark/sql/dataframe.py]: Could you also change the tests in doc string to use `True` and `False`

Comment (rxin) [inline python/pyspark/sql/dataframe.py]: then that's different from the Pandas examples?

Comment (davies) [inline python/pyspark/sql/dataframe.py]: yeah, no perfect.

**Comment by davies** [comment]
LGTM.

**Pull request closed without merging**